docs/source/best_practices.rst

```diff
@@ -0,0 +1,244 @@
+.. _best_practices:
+
+Best Practices and Performance Tips
+===================================
+
+Sort and Use Read Filtering
+---------------------------
+
+If you often read by key ranges or perform lots of logic with groups of related records, you should consider using Dask Dataframe's ``shuffle``.
```

Comment:
Might make sense to link to the dask shuffle api here https://docs.dask.org/en/stable/generated/dask.dataframe.DataFrame.shuffle.html